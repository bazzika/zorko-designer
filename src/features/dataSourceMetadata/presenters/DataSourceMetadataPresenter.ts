import {DataSourceFieldDefinition} from '../../../packages/coreTypes/DataSourceFieldDefinition';
import {BaseStatePresenter} from '../../../packages/corePresenters/BaseStatePresenter';

export type DataSourceMetadataState = {
  fields: DataSourceFieldDefinition[];
};

export class DataSourceMetadataPresenter extends BaseStatePresenter<DataSourceMetadataState> {
  static create(state?: DataSourceMetadataState) {
    return new DataSourceMetadataPresenter(state);
  }

  getDefaultState(): DataSourceMetadataState {
    return {fields: []};
  }

  setFields(fields: DataSourceFieldDefinition[]): this {
    this.state.fields = fields;
    return this;
  }
}
