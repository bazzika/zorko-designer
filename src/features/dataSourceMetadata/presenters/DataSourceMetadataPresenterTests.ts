import {DataSourceMetadataPresenter} from './DataSourceMetadataPresenter';

describe('DataSourceMetadataPresenter', () => {
  let presenter;

  beforeEach(() => {
    presenter = DataSourceMetadataPresenter.create();
  });

  it('creates default', () => {
    expect(presenter.toState()).toMatchSnapshot();
  });
});
