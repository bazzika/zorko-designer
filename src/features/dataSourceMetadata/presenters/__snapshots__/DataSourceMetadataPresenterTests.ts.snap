// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSourceMetadataPresenter creates default 1`] = `
Object {
  "fields": Array [],
}
`;

exports[`DataSourceMetadataPresenter sets new fields  1`] = `
Object {
  "fields": Array [
    Object {
      "id": "dasd",
      "levelOfMeasurement": "nominal",
      "name": "dasdsdasdasd",
      "valueType": "string",
    },
  ],
}
`;
